crates/ink/codegen/src/generator/as_dependency/contract_ref.rs

```diff
@@ -431,7 +432,6 @@ impl ContractRef<'_> {
                             .push_arg(#input_bindings)
                         )*
                     )
-                    .returns::<#ret_type>()
```

Comment:
You're gonna need to add this back so that the return type is known. You also may have to map `Self -> ()` since that's what the `InstantiateResult` implementations expect

Reply:
I just added #ret_type as a generic parameter since it must always be set. Anyway the `CreateBuilder` could do with some refactoring.

Reply:
I guess forcing devs to specify the generic is probably nicer from an error message point of view. One thing to keep in mind though is that the API should generally match that of the `call_builder`, which uses a `return()` method instead of a generic. 

We could always remove this method in the future with a bigger refactoring